examples/ivr_outbound_dialer.py

```diff
@@ -0,0 +1,359 @@
+        call.vars['playing'] = False
+        if call.vars.get('play_welcome') is True:
+            call.vars['play_welcome'] = False
+            log.info("'{}': Playing Welcome STOPPED, Lets Wait for Digits" . format(sess.uuid))
+
+    @event_callback("RECORD_START")
+    def on_record_start(self, sess):
+        call = sess.call
+        log.info("'{}': got RECORD_START ".  format(sess.uuid))
+
+    @event_callback("RECORD_STOP")
+    def on_record_stop(self, sess):
+        call = sess.call
+        log.info("'{}': got RECORD_STOP ".  format(sess.uuid))
+
+    @event_callback('CHANNEL_HANGUP')
+    def on_hangup(self, sess, job):
+        call = sess.call
+        log.info("'{}': got HANGUP ". format(sess.uuid))
+        if call.vars.get('play_welcome') is True:
+            call.vars['play_welcome'] = False
+            log.info("'{}': Got HANGUP while playing" . format(sess.uuid))
+
+
+# Make an outbound call
+# This function will be called by originator for each call
+# User is supposed to provide the outbound DID or SIP URL for each call
+#
+# SIP Call
+#   dest_url='did@domain.com'   #Remote SIP URI
+#   dest_profile='internal'     #NSG defined SIP profile name
+#   dest_endpoint='sofia'       #For SIP calls one MUST set sofia
+#
+# FreeTDM Call
+#   dest_url='[A,a]/did'        #A=ascending hunt, a=descending hunt, DID number
+#   dest_profile='g1'           #profile is used as trunk group definition: g1 == group 1
+#   endpoint='freetdm"          #For TDM calls on MUST set freetdm
+#
+# In this example we are making a FreeTDM Call.
+# Change True to False in order to make SIP calls.
+def create_url():
+    if True:
+        # Make a FreeTDM SS7/PRI Call
+        # Adding F at the end of the DID disables remote SS7 overlap dialing which can add 5 sec to the incoming call setup time
+        # Note: Developer is suppose to supply their own DID. From a list or a DB
+        return {'dest_url': 'a/1000F', 'dest_profile': 'g1', 'dest_endpoint': 'freetdm'}
+    else:
+
+        # Make a SIP Call
+        # Uncoming in order to use it.
+        return {'dest_url': '4113@10.10.12.5:6060', 'dest_profile': 'internal', 'dest_endpoint': 'sofia'}
+
+
+# Create an originator
+# Originator is a dialer
+# You can tell it how many calls to make and at what frequency
+# After the first batch of calls are complete, you can choose to start dialing again.
+# There are 3 configurable variables
+#    max_calls_per_campaign
+#    max_call_attempts_per_sec
+#    max_campaigns
+# In my example the Dialer will dial out 2 calls as par to first campaign.
+# By increasing the max_campaigns, dialer will repeat as many dial campings.
+
+max_calls_per_campaign = 1
+max_call_attempts_per_sec = 1
+max_campaigns = 1
+
+originator = get_originator([(host, port)], apps=(IVRCallLogic,), auto_duration=False, rep_fields_func=create_url)
+
+# Use place holders in order for switch to trigger create_url() function above
+# TODO: Should be hidden in switchy
```

Comment:
You can actually perform this action over the cluster fwiw. That's what `orig.pool.evals` is for. An example is in the docs: https://switchy.readthedocs.org/en/latest/callgen.html#slave-cluster

You can evaluate this `set_orig_cmd` call over all slaves in the pool.